docs/STATUS.md

```diff
@@ -0,0 +1,122 @@
+# C++ GAPIC State of the World #
```

Comment:
Can we add:
- Links to the other design docs added in this PR
- Can we link to https://aip.dev/client-libraries/4210 as the spec the project is working to?

Reply:
Done